src/core/utils/time/timeUtils.ts

```diff
@@ -20,4 +28,173 @@ function calculateRemainingTimeBreakdown(target: Date): RemainingTimeBreakdown {
   };
 }
 
-export {calculateRemainingTimeBreakdown};
+/**
```

Comment:
Added line breaks between paragraphs to improve readability

```suggestion
/**
 * Formats a Date object into a human friendly string
 *
 * Note on usage:
 *    Whenever the timezone information is not available and a date object that is passed to this 
 *    function is generated from a ISO-8601 date string (yyyy-MM-dd), `shouldShiftDateToCompensateForTimezone` option should be set to `true`.
 * 
 *    Otherwise, displayed date would be inconsistent with the data. For example, for a user with Eastern Daylight Time (to generalize, any user that has a negative UTC offset), 
 *    the following Date object is created for "2007-05-16": Tue May 15 2007 20:00:00 GMT-0400 (Eastern Daylight Time) {}. Therefore, 15 May appears on the screen. 
 *    By passing `new Date("2007-05-16")` value to `compensateForTimezone` utility, we fix this problem.
 * 
 *    When the timezone information is passed to `formatDateWithOptions`, this extra compensation is redundant as `date-fns-tz/utcToZonedTime` handles it correctly.
 *
 * @param {object} options FormatDateUtilOptions
 * @return {string} Formatted date
 */
```

Reply:
Looks great. Thanks!